pkg/controller/job/job_controller_test.go

```diff
@@ -3640,7 +3640,7 @@ func TestSyncJobWithJobSuccessPolicy(t *testing.T) {
 			wantStatus: batch.JobStatus{
 				Failed:                  1,
 				Succeeded:               1,
-				Terminating:             ptr.To[int32](0),
+				Terminating:             ptr.To[int32](2),
```

Comment:
This is a couple of changes, maybe better introduce `enableJobPodReplacementPolicy` as for other features.

Also, I would suggest to also add 2 test cases to "TestControllerSyncJob" - one with the `JobPodReplacementPolicy` feature gate disabled, and one with enabled?